docker/mqtt_metrics_collector/mqtt_metrics_collector.py

```diff
@@ -0,0 +1,89 @@
+#!/usr/bin/env python
+
+import os
+
+from prometheus_client import start_http_server, Counter
+
+# de-register default-collectors
+from prometheus_client import REGISTRY, PROCESS_COLLECTOR, PLATFORM_COLLECTOR
+REGISTRY.unregister(PROCESS_COLLECTOR)
+REGISTRY.unregister(PLATFORM_COLLECTOR)
+
+# remove python-gargage-collectior metrics
+REGISTRY.unregister(REGISTRY._names_to_collectors['python_gc_objects_uncollectable_total'])
+#REGISTRY.unregister(REGISTRY._names_to_collectors['python_gc_objects_collected_total'])
+
+import logging
+
+from base64 import b64decode
+import json
+import paho.mqtt.client as mqtt
+import random
+import urllib
+import urllib.request
+
+import sys
+
+WIS2BOX_LOGGING_LOGLEVEL = os.environ.get('WIS2BOX_LOGGING_LOGLEVEL')
+# gotta log to stdout so docker logs sees the python-logs
+logging.basicConfig(stream=sys.stdout)
+# create our own logger using same log-level as wis2box
+logger = logging.getLogger('mqtt_metrics_collector')
+logger.setLevel(WIS2BOX_LOGGING_LOGLEVEL)
+
+INTERRUPT = False
+
+def sub_connect(client, userdata, flags, rc, properties=None):
+    print("on connection to subscribe:", mqtt.connack_string(rc))
+    for s in ["xpublic/#"]:
+        client.subscribe(s, qos=1)
+
+mqtt_msg_counter = Counter('mqtt_msg_count', 'Number of messages seen on MQTT')
+mqtt_msg_by_topic_counter = Counter('mqtt_msg_count_by_topic', 'Number of messages seen on MQTT, by topic', ["topic"])
+
+def sub_mqtt_metrics(client, userdata, msg):
+    """
+      subscribe and update MQTT metrics for each new message received
+    """
+    m = json.loads(msg.payload.decode('utf-8'))
+    logger.info(f"Received message on topic ={msg.topic}")
+    mqtt_msg_by_topic_counter.labels(msg.topic).inc(1)
+    mqtt_msg_counter.inc(1)
+
+def gather_mqtt_metrics():
```

Comment:
Add function annotation and docstring (example: https://github.com/wmo-im/wis2box/blob/main/wis2box/data/__init__.py#L67-L74)